### PR TITLE
Feat/iteritems type

### PR DESCRIPTION
**Pull request by Codeprinz-1**
This adds types for `iteritems` in `Series` and `DataFrame`. I am working on a project and noticed these types are missing.

- [ ] Closes #xxxx (Replace xxxx with the Github issue number)
- [ ] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Comment by Codeprinz-1** [comment]
This adds types for `iteritems` in `Series` and `DataFrame`. I am working on a project and noticed these types are missing

File: pandas-stubs/core/series.pyi [hunk 1]
```diff
@@ -1886,6 +1886,7 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
     def set_axis(
         self, labels, *, axis: Axis = ..., copy: _bool = ...
     ) -> Series[S1]: ...
+    def iteritems(self) -> Iterable[tuple[Hashable, S1]]: ...
```
Comment (twoertwein) [inline pandas-stubs/core/series.pyi:1889]: Can you please also add two tests for these two methods?

Comment (Codeprinz-1) [reply pandas-stubs/core/series.pyi:1889]: @twoertwein  The tests have been added.

Comment (Codeprinz-1) [reply pandas-stubs/core/series.pyi:1889]: Just noticed that the tests are failing. Will fix soon

Comment (twoertwein) [reply pandas-stubs/core/series.pyi:1889]: You can also run the test locally: `poe test`

https://github.com/pandas-dev/pandas-stubs/blob/main/docs/setup.md

File: tests/test_frame.py [hunk 2]
```diff
@@ -2087,6 +2087,10 @@ def test_dict_items() -> None:
     x = {"a": [1]}
     check(assert_type(pd.DataFrame(x.items()), pd.DataFrame), pd.DataFrame)
 
+def test_iteritems() -> None:
+     df = pd.DataFrame({1: [1, 2], None: 5}, columns=pd.Index([1, None], dtype=object))
+     assert_type(df.iteritems(), Iterable[tuple[Hashable, pd.Series]] )
```
Comment (twoertwein) [inline tests/test_frame.py:2092]: Probably needs to be in a string for python 3.8: `"Iterable[tuple[Hashable, pd.Series]]"`

**Comment by Dr-Irv** [comment]
`Series.iteritems()` and `DataFrame.iteritems()` are deprecated in 1.5, and removed in version 2.0.  Use `Series.items` or `DataFrame.items` instead.

**Pull request closed without merging**